mmengine/runner/runner.py

```diff
@@ -1643,6 +1644,21 @@ def load_or_resume(self) -> None:
             self.load_checkpoint(self._load_from)
             self._has_loaded = True
 
+    def _change_log_dir(self, task_type: str) -> None:
+        """
+        Add suffix for log dir depending on task type
+        Args:
+            task_type: task type the runner lunched (train val test)
+
+        Returns:
+
+        """
+        assert task_type in ('train', 'val', 'test'), (
+            'the value of task type must be one of ["train", "val", "test"]')
+        previous_log_dir = self._log_dir
+        self._log_dir = '_'.join([self._log_dir, task_type])
+        shutil.move(previous_log_dir, self._log_dir)
```

Comment:
Hi, will the `FileHandler` in the logger still save logs to the old log file?

Reply:
Yes, I can fix this if it should modify the log file in FileHandles, and I wonder whether the save_dir in visualizer should modify in the same time.

Reply:
Yep, all opened files should be closed and be redirected to the new directory.

Reply:
If we close the files opened by `TensorboardVisBackend`, when we want to use it once again, we must to create a new instance of `TensorboardVisBackend` and it may create two events.out file in the log dir. Is this what we expect?

Reply:
Emm, resuming a visualization seems weird and dirty more or less. A more reasonable workaround is that we should lazy initialize the visualizer in `test`, `train`, `val` rather than `__init__` , but I'm not sure will it has a side effect.

Besides, if I first run `runner.train()`, and then run `runer.test()`, does it mean we need to create a new visualizer and logger during `test`? What is the lifespan of the logger and visualization? Maybe we need to rethink it carefully. What do you think of it?

Reply:
I propose the following approach: We maintain the current logic of `runner.train`, keeping the output log file names unchanged. However, we add a condition in `runner.test`. If the train_loop has already been initialized (indicating that testing is being conducted immediately after training), we will not change the log path. On the other hand, if the train_loop has not been initialized (meaning the test is being run directly), we can simply rename the folder at the end of the testing process. This way, there is no need to modify the output paths for FileHandler and Visualizer. Does this approach sound reasonable to you?